transport.go

```diff
@@ -18,6 +18,11 @@ import (
 
 var errListenerAlreadySet = errors.New("listener already set")
 
+const (
+	defaultMaxNumUnvalidatedHandshakes = 32
+	defaultMaxNumHandshakes            = 64
```

Comment:
These values seem unreasonably low.

Reply:
Maybe we should set `defaultMaxNumHandshakes` to unlimited? It's really hard to pick a limit that fits for all.

Reply:
64 still feels too low for `defaultMaxNumHandshakes`. This number is way too low to DoS for even very small machines. I think a high value > 1000 is better. I think unlimited is also better than this being < 1000.

Reply:
Changed to unlimited in 77ede93. Not entirely happy with that, but we can always change it later once we come up with a better value.